PMSCore/ProjectManager.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace PMSCore;
+
+/// <summary>
+/// Represents a Project Manager user, with permissions to manage projects and tasks.
+/// </summary>
+public class ProjectManager : User
+{
+    /// <summary>
+    /// Initializes a Project Manager with both project and task permissions.
+    /// </summary>
+    public ProjectManager(string userId)
+        : base(userId, new HashSet<Permission>
+        {
+            Permission.CanCreateProject,
+            Permission.CanChangeProjectProperty,
+            Permission.CanDeleteProject,
+            Permission.CanCreateTask,
+            Permission.CanChangeTaskProperty,
+            Permission.CanDeleteTask,
+            Permission.CanGenerateEntityReport
+        })
+    {
+    }
+
+    /// <summary>
+    /// Verifies Project Managers cannot manage users directly.
+    /// </summary>
+    public bool ValidateUserManagementRestriction()
+    {
+        return !UserPermissions.Contains(Permission.CanChangeUser) &&
+               !UserPermissions.Contains(Permission.CanDeleteUser);
```

Comment:
ProjectManager manages projects not user that is admin work I believe also it's already implmented